src/common/helpers/request.ts

```diff
@@ -22,6 +22,10 @@ client.interceptors.response.use(
       window.location.reload();
     }
 
+    if (error.response?.status === 404) {
+      window.location.href = '/#/not_found';
+    }
```

Comment:
I think this is *really* bad practice. We are stepping outside of React Router boundaries and this will fail in case we have BrowserRouter.

What we should do is have in App or somewhere global listener for the custom 404 event and then useNavigate to do navigation. 

You can then dispatch events from anywhere in the app, including Axios interceptors.

Reply:
(as an example, you can use DebouncedCombobox, there's example on dispatching/listening for the global app events).

Reply:
@beganovich Thank you for explaining the technical aspect to me. I just did it that way. Please let me know your thoughts.